pkg/reconciler/common/extensions.go

```diff
@@ -25,6 +25,7 @@ import (
 
 // Extension enables platform-specific features
 type Extension interface {
+	Append(context.Context, *mf.Manifest) error
```

Comment:
with this pr, i see that in #190 itself the interface was extedted with append . However, i think this can be avoided, as we are just adding nil implementations for all other interface implentations other than for pipeline.

can we move the append of the config.yaml to prereconcile of the OpenShift extension for TektonPipeline Reconciler. so that we can apend, and apply it in the  `pre-reconcile` of that extension.

for example: the `post-reconcile` of TektonConfig extension (to handle clean up): https://github.com/tektoncd/operator/blob/e12d59a388f11c906ae359b74f339797d29db495/pkg/reconciler/openshift/tektonconfig/extension.go#L80
and 
https://github.com/tektoncd/operator/blob/e12d59a388f11c906ae359b74f339797d29db495/pkg/reconciler/openshift/tektonconfig/extension.go#L113

Reply:
I'd defer to @vdemeester for this since this comes from #190

Reply:
If this work, I am ok with what @nikhil-thomas's suggests. I wonder why I did not take that route initially though 🤔

Reply:
@nikhil-thomas @vdemeester I've moved this to `PreReconcile`, PTAL 🤔